Makefile

```diff
@@ -247,11 +247,19 @@ ide: zephyr
 .PHONY: ashell
 ashell: zephyr
 
-# Flash Arduino 101 x86 and arc images
+# Flash for Arduino 101 x86 and arc images, olimex_stm32_e407, 96b_carbon
 .PHONY: dfu
 dfu:
+ifeq ($(BOARD), arduino_101)
 	dfu-util -a x86_app -D $(A101BIN).dfu
 	dfu-util -a sensor_core -D $(A101SSBIN).dfu
+endif
+ifeq ($(BOARD), olimex_stm32_e407)
```

Comment:
`make flash` invokes openocd tool by default for olimex_stm32_e407, but I think it makes sense to push this `dfu-util` option to Zephyr. A101 already supports flashing using JTAG or dfu-util in upstream Zephyr.